### PR TITLE
Improve saving the view state in the state file

### PR DESCRIPTION
**Pull request by mathturtle**
This saves the visibility state of the axes grid, the perspective vs.
orthographic projection mode and more camera properties (needed for
orthographic projection mode) in the state file and loads them when the
state is restored.

@cryos These are the easy bits.  Getting to the scale legend's state from the serialization code will require more refactoring.

### CODE REVIEW

File: tomviz/modules/ModuleManager.cxx [hunk 1]
```diff
@@ -496,6 +496,21 @@ bool ModuleManager::serialize(QJsonObject& doc, const QDir& stateDir,
...
+          vtkSMPropertyHelper helper(view, "InteractionMode");
+          jView["interactionMode"] = helper.GetAsInt();
+        }
+      }
+      if (view->GetProperty("AxesGrid")) {
+        vtkSMPropertyHelper helper(view, "AxesGrid");
+        vtkSMProxy* axesGridProxy = helper.GetAsProxy();
+        if (axesGridProxy) {
+          vtkSMPropertyHelper visibilityHelper(axesGridProxy, "Visibility");
+          jView["axesGridVisibility"] = visibilityHelper.GetAsInt();
```
Comment (cryos) [inline tomviz/modules/ModuleManager.cxx]: We should do an != 0 to turn into a bool in JSON.

File: tomviz/modules/ModuleManager.cxx [hunk 2]
```diff
@@ -723,6 +743,23 @@ bool ModuleManager::deserialize(const QJsonObject& doc, const QDir& stateDir)
...
+      vtkSMPropertyHelper axesGridProp(viewProxy, "AxesGrid");
+      vtkSMProxy* proxy = axesGridProp.GetAsProxy();
+      if (!proxy) {
+        vtkSMSessionProxyManager* pxm = viewProxy->GetSessionProxyManager();
+        proxy = pxm->NewProxy("annotations", "GridAxes3DActor");
+        axesGridProp.Set(proxy);
+        proxy->Delete();
+      }
+      vtkSMPropertyHelper(proxy, "Visibility")
+        .Set(view["axesGridVisibility"].toInt());
```
Comment (cryos) [inline tomviz/modules/ModuleManager.cxx]: This may then need a ternary to turn back into an int

**Comment by mathturtle** [comment]
Ok, I've switched visibility of the axes grid and isOrthographic to booleans.

**Comment by mathturtle** [comment]
This new version saves the interaction mode and the camera parallel projection state independently (as they are in the ParaView view).

**Comment by cryos** [comment]
This now works as expected, restoring the parameters in the view menu. The interaction mode is and should be separate.

**Pull request merged**